apps/emqx_s3/src/emqx_s3_schema.erl

```diff
@@ -34,11 +34,11 @@ fields(s3) ->
             )},
         {secret_access_key,
             mk(
-                string(),
+                hoconsc:union([string(), function()]),
                 #{
                     desc => ?DESC("secret_access_key"),
                     required => false,
-                    sensitive => true
+                    converter => fun secret/2
                 }
             )},
```

Comment:
Questions.
1. Is this a blessed way to deal with secrets? Are there any pitfalls?
2. Is this worth moving to `emqx_schema` module?<sup> (in a followup PR)</sup>